### ISSUE TITLE
[Package Request] - Update golang package to 1.23

### ISSUE THREAD

**Issue by mochzilla** [amazonlinux/amazon-linux-2023#818]
**What package is missing from Amazon Linux 2023? Please describe and include package name.**
go language version 1.23.2
**Is this an update to existing package or new package request?**
Existing
**Is this package available in Amazon Linux 2? If it is available via external sources such as EPEL, please specify.**
Not that I know of for this version.
**Any additional information you'd like to include. (use-cases, etc)**
Current version in AL2023 is 1.22.7.

**Comment by ozbenh** [comment]
Thanks, we are looking into it.

**Comment by Jonathan-Landeed** [comment]
Any timeline on this? It's starting to cause problems with package requirements for me. There are a lot of fundamental upgrades in 1.23, like iterators and slices.

**Comment by araman-m** [comment]
Hi, any update on this ? getting dependency error "go: go.mod requires go >= 1.23.0 (running go 1.22.7)" on AL2023

**Comment by ozbenh** [comment]
It is still being actively worked on, it should be released soon

**Comment by freeeve** [comment]
installing from source over here. y'all may as well make it 1.24 at this point.

**Comment by chadmyers** [comment]
Please keep important packages like this updated.

**Comment by Jonathan-Landeed** [comment]
If anyone else needs it in the meantime, I've been using this in `.ebextensions`

```
commands:
  01_download_go:
    command: "wget https://go.dev/dl/go1.23.4.linux-amd64.tar.gz -O /tmp/go.tar.gz"
    ignoreErrors: false
  02_remove_old_go:
    command: "sudo rm -rf /usr/local/go"
    ignoreErrors: false
  03_install_go:
    command: "sudo tar -C /usr/local -xzf /tmp/go.tar.gz"
    ignoreErrors: false
  04_set_path:
    command: "sudo ln -sf /usr/local/go/bin/go /usr/bin/go"
    ignoreErrors: false
```

**Comment by jay-babu** [comment]
@ozbenh no, it has not. This is very easy to verify: https://docs.aws.amazon.com/linux/al2023/release-notes/all-packages-AL2023.6.html. This issue should be reopened

![Image](https://github.com/user-attachments/assets/e6e3ed59-cf6a-45d2-8df0-ff5e6291c08a)

**Comment by chadmyers** [comment]
According to this, it's there (1.24).

https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.6.20250303.html

I just pulled `public.ecr.aws/amazonlinux/amazonlinux:latest` and ran `dnf install -y -q golang` and then:

```
bash-5.2# go version
go version go1.24.0 linux/amd64
```

**Comment by ozbenh** [comment]
@jay-babu it's definitely there, I am not sure why that page doesn't reflect it

**Comment by ozbenh** [comment]
It does look like this page doesn't have all the updates corresponding to the latest releases it claims to show (it's also missing `clamav1.4` for example. We are investigating what's going on there

**Comment by sktomer** [comment]
Here is what I think happened:

* The link https://docs.aws.amazon.com/linux/al2023/release-notes/all-packages-AL2023.6.html used to show all the package updates in the pilot `2023.6` release. This corresponds to https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.6.20241010.html which shows golang version `1.22.7-1.amzn2023.0.1`.
* So even when golang was updated in subsequent `2023.6.20250303` release: https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.6.20250303.html, the main page still showed the pilot release's package versions.
* The takeaway here should be to not rely on the minor version release notes, but to actually look at the release notes from latest system-release version, example: https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.6.20250317.html
